bin/exec.go

```diff
@@ -85,93 +77,49 @@ func main() {
 			log.Infoln("Not Executing the Experiment : " + engineDetails.Experiments[i])
 			break
 		}
+		log.Info("Printing the ENV Variables", experiment.Env)
 
-		log.Infoln("Getting the Default ENV Variables")
-
-		// Get the Default ENV's from ChaosExperiment
-		experiment.SetDefaultEnv(clients)
-
-		log.Info("Printing the Default Variables", experiment.Env)
-
-		// Get the ConfigMaps for patching them in the job creation
-		log.Infoln("Find the configMaps in the chaosExperiments")
-
-		experiment.SetConfigMaps(clients)
-
-		err := experiment.ValidateConfigMaps(clients)
+		// Patch ConfigMaps to ChaosExperiment Job
+		err := experiment.PatchConfigMaps(clients)
 		if err != nil {
-			log.Infof("Aborting Execution")
-			return
+			break
 		}
-
-		experiment.SetSecrets(clients)
-		err = experiment.ValidateSecrets(clients)
+		// Patch Secrets to ChaosExperiment Job
+		err = experiment.PatchSecrets(clients)
 		if err != nil {
-			log.Infof("Aborting Execution")
-			return
+			break
 		}
-
-		log.Infof("Validated ConfigMaps: %v", experiment.ConfigMaps)
-		log.Infof("Validated Secrets: %v", experiment.Secrets)
-
 		// Adding VolumeBuilders, according to the ConfigMaps, and Secrets from ChaosEXperiment
 		experiment.VolumeOpts.VolumeBuilders = utils.CreateVolumeBuilder(experiment.ConfigMaps, experiment.Secrets)
 
 		// Adding VoulmeMounts, according to the configMaps, and Secrets from ChaosEXperiment
 		experiment.VolumeOpts.VolumeMounts = utils.CreateVolumeMounts(experiment.ConfigMaps, experiment.Secrets)
 
-		// OverWriting the Defaults Varibles from the ChaosEngine ENV
-		log.Infoln("Patching some required ENV's")
-		experiment.SetEnvFromEngine(engineDetails.Name, clients)
-
-		// Adding some addition necessary ENV's
-		experiment.Env["CHAOSENGINE"] = engineDetails.Name
-		experiment.Env["APP_LABEL"] = engineDetails.AppLabel
-		experiment.Env["APP_NAMESPACE"] = engineDetails.AppNamespace
-		experiment.Env["APP_KIND"] = engineDetails.AppKind
-
-		log.Info("Printing the Over-ridden Variables")
-		log.Infoln(experiment.Env)
-
-		log.Infoln("Getting all the details of the experiment Name : " + engineDetails.Experiments[i])
-
-		// Fetching more details from the ChaosExperiment needed for execution
-		experiment.SetImage(clients)
-		experiment.SetArgs(clients)
-		experiment.SetLabels(clients)
-		log.Infof("Variables for ChaosJob: Experiment Labels: %v, Experiment Image: %v, experiment.Args: %v", experiment.ExpLabels, experiment.ExpImage, experiment.ExpArgs)
-
-		// Generation of Random String for appending it into Job
-		randomString := utils.RandomString()
-
-		// Setting the JobName in Experiment Realted struct
-		experiment.JobName = engineDetails.Experiments[i] + "-" + randomString
-
-		log.Infof("JobName for this Experiment : %v", experiment.JobName)
-
 		// Creation of PodTemplateSpec, and Final Job
 		err = utils.DeployJob(experiment, clients)
 		if err != nil {
 			log.Infof("Error while building Job : %v", err)
```

Comment:
Unable to construct chaos experiment job due to:

Reply:
Done.